anomalib/data/__init__.py

```diff
@@ -30,61 +31,24 @@ def get_datamodule(config: Union[DictConfig, ListConfig]) -> LightningDataModule
 
-    if config.dataset.format.lower() == "mvtec":
-        datamodule = MVTec(
-            # TODO: Remove config values. IAAALD-211
-            root=config.dataset.path,
-            category=config.dataset.category,
-            image_size=(config.dataset.image_size[0], config.dataset.image_size[1]),
-            train_batch_size=config.dataset.train_batch_size,
-            test_batch_size=config.dataset.test_batch_size,
-            num_workers=config.dataset.num_workers,
-            seed=config.project.seed,
-            task=config.dataset.task,
-            transform_config_train=config.dataset.transform_config.train,
-            transform_config_val=config.dataset.transform_config.val,
-            create_validation_set=config.dataset.create_validation_set,
-        )
-    elif config.dataset.format.lower() == "btech":
-        datamodule = BTech(
-            # TODO: Remove config values. IAAALD-211
-            root=config.dataset.path,
-            category=config.dataset.category,
-            image_size=(config.dataset.image_size[0], config.dataset.image_size[1]),
-            train_batch_size=config.dataset.train_batch_size,
-            test_batch_size=config.dataset.test_batch_size,
-            num_workers=config.dataset.num_workers,
-            seed=config.project.seed,
-            task=config.dataset.task,
-            transform_config_train=config.dataset.transform_config.train,
-            transform_config_val=config.dataset.transform_config.val,
-            create_validation_set=config.dataset.create_validation_set,
-        )
-    elif config.dataset.format.lower() == "folder":
-        datamodule = Folder(
-            root=config.dataset.path,
-            normal_dir=config.dataset.normal_dir,
-            abnormal_dir=config.dataset.abnormal_dir,
-            task=config.dataset.task,
-            normal_test_dir=config.dataset.normal_test_dir,
-            mask_dir=config.dataset.mask,
-            extensions=config.dataset.extensions,
-            split_ratio=config.dataset.split_ratio,
-            seed=config.project.seed,
-            image_size=(config.dataset.image_size[0], config.dataset.image_size[1]),
-            train_batch_size=config.dataset.train_batch_size,
-            test_batch_size=config.dataset.test_batch_size,
-            num_workers=config.dataset.num_workers,
-            transform_config_train=config.dataset.transform_config.train,
-            transform_config_val=config.dataset.transform_config.val,
-            create_validation_set=config.dataset.create_validation_set,
-        )
-    else:
-        raise ValueError(
-            "Unknown dataset! \n"
-            "If you use a custom dataset make sure you initialize it in"
-            "`get_datamodule` in `anomalib.data.__init__.py"
-        )
+    # Since in the new config based on LightningCLI, seed_everything is a separate parameter, we set the one in dataset
+    # to point to it.
+    if config.data.init_args.seed == 0:
```

Comment:
This check is same as the main. It is just to warn the users that if they set seed to 0, they will see a different behaviour than the previous versions.

Reply:
I don't follow. I thought this was to ensure that the dataset seed is the same as the global seed in the case where the user does not provide a specific seed for the dataset (which is actually not necessary, because just setting a global seed will suffice to ensure consistent splits between runs).

Let's say the user wants random training runs, but consistent subset splits between runs. In this case the user will set `seed_everything` to `null`, and `data.seed` to a fixed integer. If the user picks 0 as dataset seed, the seed will be overwritten by `None` as a result of this check, which silently disables deterministic subset splitting.

Besides, it's not necessary to warn the users here, because the `dataset.seed` parameter was previously not exposed to the users. So I think we can safely remove this check altogether.